src/main/java/org/purescript/psi/imports/PSImportDeclarationImpl.kt

```diff
@@ -140,6 +142,54 @@ class PSImportDeclarationImpl(node: ASTNode) : PSPsiElement(node) {
             PSImportedData::class.java
         )
 
+    /**
+     * @return the [PSNewTypeConstructor] elements imported by this declaration
+     */
+    val importedNewTypeConstructors: List<PSNewTypeConstructor>
+        get() {
+            val importedModule = importedModule ?: return emptyList()
+            val exportedNewTypeConstructors = importedModule.exportedNewTypeConstructors
+
+            val importedItems = importList?.importedItems
+                ?: return exportedNewTypeConstructors
+
+            val importedNewTypeConstructors = mutableListOf<PSNewTypeConstructor>()
+            val importedDataElements = importedItems.filterIsInstance<PSImportedData>()
+            if (isHiding) {
+                /*
+                 * Partially hiding imported data does not work in an intuitive way.
+                 * Here are some examples using Data.Maybe:
+                 *
+                 *   import Data.Maybe hiding (Maybe(..))             -- Hides type and constructors
+                 *   import Data.Maybe hiding (Maybe)                 -- Hides nothing
+                 *   import Data.Maybe hiding (Maybe())               -- Hides nothing
+                 *   import Data.Maybe hiding (Maybe(Just))           -- Hides nothing
+                 *   import Data.Maybe hiding (Maybe(Just, Nothing))  -- Hides nothing
+                 *   import Data.Maybe hiding (Maybe(Nothing, Just))  -- Hides type and constructors
+                 *
+                 * TODO
+                 *  I'll account for the Maybe(..) case for now, and ignore the rest.
+                 *  It's definitely wrong, but I'd rather the code be simple and wrong
+                 *  than complicated and still somehow wrong.
+                 */
```

Comment:
There's an issue about this in the purescript compiler repo: https://github.com/purescript/purescript/issues/3424